src/components/Header.jsx

```diff
@@ -14,41 +14,50 @@ export const Header = () => {
   const cartCount = cartToShow.reduce((acc, curr) => (acc = acc + curr.qty), 0);
 
   const navigate = useNavigate();
-  const location = useLocation();
-  const { productReducerDispatch, productReducerState } = useProducts();
-  useEffect(
-    () =>
-      productReducerState.filterBySearch.trim() &&
-      location.pathname !== "/products" &&
-      navigate("/products"),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [productReducerState.filterBySearch]
-  );
+  const { productsFromDB } = useProducts();
+  const searchOptions = productsFromDB.map((item) => ({
+    ...item,
+    label: item.name + " by " + item.author,
+  }));
+
+  const { theme, toggleDarkMode } = useTheme();
 
   return (
     <header>
       <div className="navigation-container">
         <Link to="/">
           <h2>CobraStore</h2>
         </Link>
-        <input
-          value={productReducerState.filterBySearch}
-          type="text"
-          placeholder="Search"
-          className="nav-search search-desktop"
-          onChange={(e) =>
-            productReducerDispatch({
-              type: "FILTER_BY_SEARCH",
-              payload: e.target.value,
-            })
-          }
-        />
+        <div className="nav-search search-desktop search-wrapper">
+          <Select
+            options={searchOptions}
+            isClearable="true"
+            placeholder="Search"
+            onChange={(opt) =>
+              opt && navigate(`/products/${opt?._id}`, { replace: true })
+            }
+          />
+        </div>
         <div className="nav-right">
           {isUserLoggedIn && (
-            <button className="btn btn-secondary" onClick={logoutHandler}>
+            <button
+              className="btn btn-secondary logout-btn"
+              onClick={logoutHandler}
+            >
               Logout
             </button>
           )}
+          {theme === "light" ? (
+            <i
+              className="fas fa-moon fa-3x theme-toggle"
+              onClick={() => toggleDarkMode()}
+            />
+          ) : (
+            <i
+              className="fas fa-sun fa-3x theme-toggle"
+              onClick={() => toggleDarkMode()}
+            />
+          )}
```

Comment:
suggestion: you could have kept just one `<i>` here and just changed the classname

Reply:
Done